packages/thirdweb/src/react/web/ui/TransactionButton/TransactionModal.tsx

```diff
@@ -37,14 +37,19 @@ export function TransactionModal(props: ModalProps) {
   useQuery({
     queryKey: ["transaction-modal-event"],
     queryFn: () => {
-      if (!account || !wallet) return;
+      // Query data cannot be undefined, return null
+      if (!account || !wallet) {
+        return null;
```

Comment:
should it just throw instead so it retries later? 

also should we add !!account to enabled prop below?

Reply:
updated